### PR TITLE
feat: add severity level field to the support modal

### PR DESCRIPTION
**Pull request by MarconLP**
## Problem

add severity level field to the support modal #19781
2024-01-17 at 16 38 11@2x

## How did you test this code?

manually sent a ticket through the app and check if the ticket contains the correct `severity_level`

### CODE REVIEW

File: frontend/src/lib/components/Support/SupportForm.tsx [hunk 1]
```diff
@@ -76,6 +85,19 @@ export function SupportForm(): JSX.Element | null {
...
+                <LemonSelect
+                    fullWidth
+                    options={Object.entries(SEVERITY_LEVEL_TO_NAME).map(([key, value]) => ({
+                        label: value,
+                        value: key,
+                    }))}
+                />
+            </Field>
+            <span className="text-muted cursor-pointer">
+                Checkout the{' '}
```
Comment (corywatilo) [inline frontend/src/lib/components/Support/SupportForm.tsx:98]: Check out

File: frontend/src/lib/components/Support/supportLogic.ts [hunk 2]
```diff
@@ -70,13 +70,21 @@ export const TARGET_AREA_TO_NAME = {
     'posthog-3000': 'PostHog 3000',
 }
 
+export const SEVERITY_LEVEL_TO_NAME = {
+    critical: 'Critical - Outage / Data loss / Breach',
```
Comment (corywatilo) [inline frontend/src/lib/components/Support/supportLogic.ts:74]: Sentence case these, from:

> Critical - Outage / Data loss / Breach

to 

> Critical - Outage / data loss / breach

etc

**Review by corywatilo — COMMENTED** [review]
Didn't test, just looked at copy etc

I'd also maybe consider moving the field _below_ topic and/or the actual message input.

File: frontend/src/lib/components/Support/supportLogic.ts (hunk 2 above)
Comment (MarconLP) [reply frontend/src/lib/components/Support/supportLogic.ts:74]:
```suggestion
    critical: 'Critical - Outage / data loss / breach',
```

File: frontend/src/lib/components/Support/SupportForm.tsx (hunk 1 above)
Comment (MarconLP) [reply frontend/src/lib/components/Support/SupportForm.tsx:98]:
```suggestion
                Check out the{' '}
```

File: frontend/src/lib/components/Support/SupportForm.tsx [hunk 3]
```diff
@@ -43,6 +43,7 @@ export function SupportForm(): JSX.Element | null {
     const { objectStorageAvailable } = useValues(preflightLogic)
     // the support model can be shown when logged out, file upload is not offered to anonymous users
     const { user } = useValues(userLogic)
+    const supportRequestKind = useSelector(supportLogic.selectors.kind)
```
Comment (neilkakkar) [inline frontend/src/lib/components/Support/SupportForm.tsx:46]: you don't need to do this separately, on line 41 you can just do:

```
const { sendSupportRequest, kind } = useValues(supportLogic)
```

Selectors are values too!

Comment (neilkakkar) [reply frontend/src/lib/components/Support/SupportForm.tsx:46]: oh, it's also just a part of sendSupportRequest it seems, so you don't really need the selector at all?

**Review by neilkakkar — APPROVED** [review]
One thing to fix, but otherwise LGTM!

File: frontend/src/lib/components/Support/SupportForm.tsx (hunk 3 above)
Comment (MarconLP) [reply frontend/src/lib/components/Support/SupportForm.tsx]: removed the selector

**Pull request merged**